### ISSUE TITLE
Generate proper constructors for fixed-size arrays

### ISSUE THREAD

**Issue by SaculRennorb** [Ultz/Silk.NET#214]
# Summary
as you can see here
![image](https://user-images.githubusercontent.com/18741506/86232293-01ebb580-bb94-11ea-8e09-5d7044e9cca7.png)
the ctor for `ClearColorValue` does not work with float values. 
this stems from the fact that some of the fields occupy the the same location, but get initialized with, for example `uint_0 = default`, _AFTER_ other fields at the same location, essentially overriding them with `default`.
for me it works when i use the last fields (the uint ones) but for everything else the values get zeroed out.

# Steps to reproduce
- Platform: Desktop
- Framework Version: .NET Core 3
i don't know if this is platform specific, i'm not that familiar with the net core spec

# _A_ solution
 would be to split up the ctor into separate ones that initialize base and then override the correct fields:
```C#
 public ClearColorValue(float float32_0 = default, float float32_1 = default, float float32_2 = default, float float32_3 = default) 
  : this() {
  Float32_0 = float32_0;
  Float32_1 = float32_1;
  Float32_2 = float32_2;
  Float32_3 = float32_3;
 }
public ClearColorValue(int int32_0 = default, int int32_1 = default, int int32_2 = default, int int32_3 = default) 
 : this() {
  Int32_0 = int32_0;
  Int32_1 = int32_1;
  Int32_2 = int32_2;
  Int32_3 = int32_3;
}
public ClearColorValue(uint uint32_0 = default, uint uint32_1 = default, uint uint32_2 = default, uint uint32_3 = default) 
 : this() {
  Uint32_0 = uint32_0;
  Uint32_1 = uint32_1;
  Uint32_2 = uint32_2;
  Uint32_3 = uint32_3;
}
```

i didn't create a pr for two main reasons:
 - this is more of a design decision, i don't want to take that away
 - i suspect this ctor is generated, if so i guess it wouldn't even as simple as just changing it in a pr. again: i don't know too much how this project works under the hood, i just know suff about c#

another solution would be to do some pointer arithmetic in the ctors so you don't need to invoke the empty ctor, but i think the first suggestion would be cleaner.

btw this took me ~5h to figure out with RenderDec ...  was the last thing i suspected.

ofcourse you can always just use it the c way by first calling the empty ctor and then setting the fields, but i prefer a propper ctor.

**Comment by Perksey** [comment]
Yeah our treatment of unions and structs contianing fixed-size arrays hasn't always been great. Luckily in 2.0 I intend to implement #163 which will make using these types a lot nicer, and I will try and make sure to implement an appropriate constructor for it as well. In the meantime, you'll probably just have to use the empty ctor. Sorry about this :( it'll be fixed in the next breaking release.

**Comment by SaculRennorb** [comment]
ah well, glad you at least know about this and are planing on doing something to fix it.

**Comment by Perksey** [comment]
Resolved in #340, will be part of 2.0 Preview 3. All struct constructor parameters are nullable, and fields will only be set if you pass in a value that is not null. Example:
```cs
namespace Silk.NET.Vulkan
{
    [NativeName("Name", "VkSamplerCustomBorderColorCreateInfoEXT")]
    public unsafe partial struct SamplerCustomBorderColorCreateInfoEXT
    {
        public SamplerCustomBorderColorCreateInfoEXT
        (
            StructureType? sType = StructureType.SamplerCustomBorderColorCreateInfoExt,
            void* pNext = null,
            ClearColorValue? customBorderColor = null,
            Format? format = null
        ) : this()
        {
            if (sType is not null)
            {
                SType = sType.Value;
            }

            if (pNext is not null)
            {
                PNext = pNext;
            }

            if (customBorderColor is not null)
            {
                CustomBorderColor = customBorderColor.Value;
            }

            if (format is not null)
            {
                Format = format.Value;
            }
        }

/// <summary></summary>
        [NativeName("Type", "VkStructureType")]
        [NativeName("Type.Name", "VkStructureType")]
        [NativeName("Name", "sType")]
        public StructureType SType;
/// <summary></summary>
        [NativeName("Type", "void*")]
        [NativeName("Type.Name", "void")]
        [NativeName("Name", "pNext")]
        public void* PNext;
/// <summary></summary>
        [NativeName("Type", "VkClearColorValue")]
        [NativeName("Type.Name", "VkClearColorValue")]
        [NativeName("Name", "customBorderColor")]
        public ClearColorValue CustomBorderColor;
/// <summary></summary>
        [NativeName("Type", "VkFormat")]
        [NativeName("Type.Name", "VkFormat")]
        [NativeName("Name", "format")]
        public Format Format;
    }
}
```
